forc/src/utils/restricted_names.rs

```diff
@@ -93,36 +94,34 @@ pub fn is_glob_pattern<T: AsRef<str>>(name: T) -> bool {
     name.as_ref().contains(&['*', '?', '[', ']'][..])
 }
 
-#[test]
-fn test_invalid_char() {
-    assert_eq!(
-        contains_invalid_char("test#proj", "package name"),
-        std::result::Result::Err(
-            "invalid character `#` in package name: `test#proj`, \
-                characters must be Unicode XID characters \
-                (numbers, `-`, `_`, or most letters)"
-                .into()
-        )
-    );
-    assert_eq!(
-        contains_invalid_char("test proj", "package name"),
-        std::result::Result::Err(
-            "invalid character ` ` in package name: `test proj`, \
-                characters must be Unicode XID characters \
-                (numbers, `-`, `_`, or most letters)"
-                .into()
-        )
-    );
-    assert_eq!(
-        contains_invalid_char("", "package name"),
```

Comment:
and here 
`contains_invalid_char("", "package name").map_err(|e| e.to_string())`